osu.Game/Overlays/RankingsOverlay.cs

```diff
@@ -100,20 +110,31 @@ protected override void LoadComplete()
                 if (Country.Value != null)
                     Scope.Value = RankingsScope.Performance;
 
-                Scheduler.AddOnce(loadNewContent);
+                if (Scope.Value != RankingsScope.Spotlights)
+                    Scheduler.AddOnce(loadNewContent);
```

Comment:
I am *very* not fine with this way of fixing the problem. What the crash scenario demonstrated is that the overlay can be forced to fire off requests to fetch spotlight ranks while not being in a ready state. At the point of the nullref being thrown neither `header.Spotlight` nor `header.Spotlights` were initialised to sane values.

This check is essentialy monkey-patching this edge case. I had to spend a good 10 minutes to understand why this is making the crash go away, and it is due to:

* spotlights changing `Country.Value` to `null`,
* which fires this `Country.BindValueChanged` subscriber with `RankingsScope.Spotlights` and not-ready state of not having any spotlights at all.

I feel like letting this remain at this state is just asking for trouble further down the line. In my opinion the proper direction to go forward with this is to go back on the decision to add the spotlight selector to the header and make a separate component for spotlight rankings, which will have:

* the spotlight dropdown along with the extra content,
* the actual ranking table,
* and the beatmap covers in the footer.

This component would be solely responsible for firing its own request and making sure it's in a sane state to start filtering, therefore encapsulating the spotlight logic and removing clutter from the top-level overlay.

Reply:
Refactored